python/lsst/ctrl/bps/bps_core.py

```diff
@@ -300,78 +277,80 @@ def _create_science_graph(self):
 
         self.sci_graph = networkx.DiGraph()
-        ncnt = 0
-        tcnt = 0
-        dcnt = 0
+        tcnt = 0   # task node counter
+        dcnt = 0   # dataset ref node counter
 
         dsname_to_node_id = {}
-        self.qgnodes = {}
-        pipeline = []
-        for task_id, nodes in enumerate(self.qgraph):
-            _LOG.debug(task_id)
-            task_def = nodes.taskDef
-            pipeline.append(task_def.label)
+
+        # Using a dictionary to get an ordered "set" of
+        # pipeline task labels.  For efficiency reasons,
+        # creating this while traversing QuantumGraph,
+        # instead of re-traversing science graph later.
+        pipeline_task_labels = {}
+
+        for node in self.qgraph:
+            _LOG.debug("type(node)=%s", type(node))
+            _LOG.debug("nodeId=%s", node.nodeId)
+
+            task_def = node.taskDef
+            pipeline_task_labels[task_def.label] = True
 
             _LOG.debug("config=%s", task_def.config)
             _LOG.debug("taskClass=%s", task_def.taskClass)
             _LOG.debug("taskName=%s", task_def.taskName)
             _LOG.debug("label=%s", task_def.label)
-            for quantum in nodes.quanta:
-                _LOG.debug("actualInputs=%s", quantum.actualInputs)
-                _LOG.debug("id=%s", quantum.id)
-                _LOG.debug("run=%s", quantum.run)
-                _LOG.debug("initInputs=%s", quantum.initInputs)
-                ncnt += 1
-                tcnt += 1
-                # tnode_name = "task%d (%s)" % (ncnt, task_def.taskName)
-                tnode_name = "%06d" % (ncnt)
-                self.sci_graph.add_node(
-                    tnode_name,
-                    node_type=TASKNODE,
-                    task_def_id=task_id,
-                    task_abbrev=task_def.label,
-                    shape="box",
-                    fillcolor="gray",
-                    # style='"filled,bold"',
-                    style="filled",
-                    label=".".join(task_def.taskName.split(".")[-2:]),
-                )
-                quanta2 = [quantum]
-                self.qgnodes[tnode_name] = QuantumGraphTaskNodes(task_def, quanta2, quantum.initInputs, {})
-
-                # Make nodes for inputs
-                for ds_refs in quantum.predictedInputs.values():
-                    for ds_ref in ds_refs:
-                        ds_name = "%s+%s" % (ds_ref.datasetType.name, ds_ref.dataId)
-                        if ds_name not in dsname_to_node_id:
-                            ncnt += 1
-                            dcnt += 1
-                            dsname_to_node_id[ds_name] = ncnt
-                        fnode_name = "%06d" % dsname_to_node_id[ds_name]
-                        fnode_desk = pretty_dataset_label(ds_name)
+
+            tcnt += 1
+
+            tnode_name = "%06d" % (node.nodeId.number)
+            self.sci_graph.add_node(
+                tnode_name,
+                node_type=TASKNODE,
```

Comment:
The graph node property names should be changing/growing in a future ticket, so leaving this alone here.  But I do want to clarify your comment independent of what the code is or isn't doing.  I thought a Quantum was defined as the task (specific config, etc) + data it will process.